senders/mattermost/sender_test.go

```diff
@@ -0,0 +1,72 @@
+package mattermost_test
+
+import (
+	"testing"
+
+	"github.com/moira-alert/moira/senders/mattermost"
+
+	logging "github.com/moira-alert/moira/logging/zerolog_adapter"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestInit(t *testing.T) {
+	logger, _ := logging.ConfigureLog("stdout", "debug", "test", true)
+	Convey("Init tests", t, func() {
+		sender := &mattermost.Sender{}
+
+		Convey("Empty url", func() {
+			senderSettings := map[string]string{"api_token": "qwerty", "front_uri": "qwerty"}
+			err := sender.Init(senderSettings, logger, nil, "")
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("Empty api_token", func() {
+			senderSettings := map[string]string{"url": "qwerty", "front_uri": "qwerty"}
+			err := sender.Init(senderSettings, logger, nil, "")
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("Empty front_uri", func() {
+			senderSettings := map[string]string{"url": "qwerty", "api_token": "qwerty"}
+			err := sender.Init(senderSettings, logger, nil, "")
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("Full config", func() {
+			senderSettings := map[string]string{"url": "qwerty", "api_token": "qwerty", "front_uri": "qwerty"}
+			err := sender.Init(senderSettings, logger, nil, "")
+			So(err, ShouldBeNil)
+		})
+	})
+}
+
+//TestSender is integration test, run it manually with your Url, Token and Channel ID.
```

Comment:
? :)

Reply:
It is small integration test, you should run it manually. I comment it because it works only with right credentials (your Url, Token and Channel ID). You can run https://hub.docker.com/r/mattermost/mattermost-preview locally, create bot, uncomment test and run it.